examples/medplum-provider/src/components/tasks/TaskListItem.tsx

```diff
@@ -1,45 +1,54 @@
-import { Group, Stack, Text } from '@mantine/core';
-import { formatDate, formatHumanName, getDisplayString } from '@medplum/core';
+import { Anchor, Group, Stack, Text } from '@mantine/core';
+import { formatDate, formatHumanName } from '@medplum/core';
 import { HumanName, Task } from '@medplum/fhirtypes';
-import { JSX } from 'react';
+import { JSX, MouseEvent } from 'react';
 import classes from './TaskListItem.module.css';
 import cx from 'clsx';
 import { useResource } from '@medplum/react';
+import { useNavigate } from 'react-router-dom';
 
 interface TaskListItemProps {
   task: Task;
   selectedTask: Task | undefined;
-  onClick: () => void;
 }
 
 export function TaskListItem(props: TaskListItemProps): JSX.Element {
-  const { task, selectedTask, onClick } = props;
+  const { task, selectedTask } = props;
+  const navigate = useNavigate();
   const isSelected = selectedTask?.id === task.id;
   const patient = useResource(task.for);
   const owner = useResource(task.owner);
 
+  const handleClick = (e: MouseEvent<HTMLAnchorElement>): void => {
+    if (e.button === 0) {
+      e.preventDefault();
+      navigate(`${task.id}`)?.catch(console.error);
+    }
+  };
+
   return (
-    <Group
-      p="xs"
-      align="center"
-      wrap="nowrap"
-      className={cx(classes.contentContainer, {
-        [classes.selected]: isSelected,
-      })}
-      onClick={onClick}
-    >
-      <Stack gap={0}>
-        <Text fw={700} className={classes.content}>
-          {getDisplayString(task)}
-        </Text>
-        {task.restriction?.period && <Text fw={500}>Due {formatDate(task.restriction?.period?.end)}</Text>}
-        {patient?.resourceType === 'Patient' && <Text>For: {formatHumanName(patient.name?.[0] as HumanName)}</Text>}
-        {owner?.resourceType === 'Practitioner' && (
-          <Text size="sm" c="dimmed">
-            Assigned to {formatHumanName(owner.name?.[0] as HumanName)}
+    <Anchor href={`${task.id}`} onClick={handleClick} td="none" c="inherit">
```

Comment:
Could this use `MedplumLink` instead to take care of the click handling and navigation?